lib/phonegap/cordova-dependence.js

```diff
@@ -0,0 +1,112 @@
+ */
+module.exports.exec = function(projectDir, extEvents) {
+    //store existing working directory
+    var cwd = process.cwd();
+    extEvents = extEvents ? extEvents : ConsoleEmitter(extEvents);
+    var PROJECTROOT;
+    return Q()
+        .then(function() {
+            var project = getProjectRoot(projectDir);
+            if (!project) {
+                return Q.reject(new Error('"' + projectDir + '" ' + 'does not point to a valid PhoneGap project'));
+            } else {
+                PROJECTROOT = project;
+                return project;
+            }
+        }).then(function(projectRoot) {
+            //change current working directory
+            shell.cd(projectRoot);
+            var config,
+                pkgJson,
+                pkgJsonPath = path.resolve(projectRoot, 'package.json');
+
+            if (!fs.existsSync(pkgJsonPath)) {
+                extEvents.emit('warn', 'No package.json was found for your project. Creating one from config.xml');
+                config = new ConfigParser(path.resolve(projectRoot, 'config.xml'));
+                pkgJson = {};
+                if (config.name()) {
+                    pkgJson.displayName = config.name();
+                }
+                // Pkjson.name should equal config's id.
+                if(config.packageName()) {
+                    pkgJson.name = config.packageName().toLowerCase();
+                } else if(!config.id()) {
+                    pkgJson.name = path.basename(projectRoot).toLowerCase().replace(' ', '') || 'helloworld';
+                }
+
+                pkgJson.version = config.version() || '1.0.0';
+                fs.writeFileSync(pkgJsonPath, JSON.stringify(pkgJson, null, 4), 'utf8');
+
+                return pkgJsonPath;
+            } else {
+                return pkgJsonPath;
+            }
+        }).then(function(pkgJsonPath) {
+            pkgJson =  path.relative(__dirname, pkgJsonPath);
+            pkgJson = require(pkgJson);
+
+            //Does not have cordova dependency in package.json
+            if (!pkgJson.dependencies || !pkgJson.dependencies.hasOwnProperty('cordova')) {
+                if (!shell.which('npm')) {
+                    return Q.reject(new Error('"npm" command line tool is not installed: make sure it is accessible on your PATH.'));
+                }
+                var deferred = Q.defer();
+                // Find the latest version of Cordova published and install it
+                var cordovaCommand = 'npm install cordova --save';
+                extEvents.emit('log', 'Adding Cordova dependency with: "' + cordovaCommand + '"');
+                extEvents.emit('log', 'This may take a few minutes ...');
+                var child = shell.exec(cordovaCommand, {
+                    silent: true
+                }, function(code, stdout, stderr) {
+                    if (code != 0) {
+                        e = new Error('Error from npm: ' + stdout.replace('\n', ''));
```

Comment:
Another way to frame this question: in the case that `npm install cordova --save` fails, is valuable output spit out in `stdout` or in `stderr` ?